test.py

```diff
@@ -412,6 +412,16 @@ def __init__(self, test_no, shortname, suite):
         # Unique file name, which is also readable by human, as filename prefix
         self.uname = "{}.{}".format(self.shortname, self.id)
         self.log_filename = os.path.join(suite.options.tmpdir, self.mode, self.uname + ".log")
+        Test._reset(self)
+
+    def reset(self):
```

Comment:
Wouldn't you achieve the same result by simply creating the test again? As you say in the commit message, reset is similar to init - so it's as if you dropped your previous Test object and created it again. Is there a difference?

Reply:
there is a bigger difference. A re-tried flaky test is still the same test. It has a history of runs and how many times it was retried, in particular. It's displayed as FLKY  in the output. The reset is only resetting the status fields, and fields which are overwritten by running the test, but there is a lot of content that needs to be prepared once. Plus of course I do not want to extend the set of tests to run as I run them, that would complicate cancellation